graylog2-server/src/main/java/org/graylog/security/permissions/GRNPermission.java

```diff
@@ -33,7 +34,14 @@ public static GRNPermission create(String type, GRN target) {
 
     @Override
     public boolean implies(Permission p) {
-        // By default only supports comparisons with other GRNPermission
+        // The admin permission also implies access to every GRNPermission
+        if ((p instanceof WildcardPermission)) {
```

Comment:
This doesn't work, unfortunately. The permission `p` that is passed in is the `entity:own` `GRNPermission`: `GRNPermission{type=entity:own, target=grn::::stream:000000000000000000000003}`

That would mean we would need this check in the `WildcardPermission` class instead. (which we cannot do because it's part of shiro)

I am afraid we need to find a different solution.

Reply:
You're right. It only worked because I was testing it with the admin user itself, not with a regular user that has the admin role.